tests/samples/VarDefine.scala

```diff
@@ -0,0 +1,12 @@
+package org.scalanative.bindgen.samples
+
+import scala.scalanative._
+import scala.scalanative.native._
+
+@native.link("bindgentests")
+@native.extern
+object VarDefine {
+  @name("a")
+  def A: native.CInt = native.extern
```

Comment:
I'd expect this be a `var` since it is not `extern cont int a`?

Reply:
Thanks! Yes, it at least should be `val`
I tried to replace it with `var` and change value of `A` but got compiler error
It is not informative:
```sbt
[error] scala.scalanative.util.UnsupportedException: List(10) (class scala.collection.immutable.$colon$colon)
[error] 	at scala.scalanative.util.package$.unsupported(package.scala:17)
[error] 	at scala.scalanative.nscplugin.NirGenExpr$ExprBuffer.genApplyExternAccessor(NirGenExpr.scala:1572)
[error] 	at scala.scalanative.nscplugin.NirGenExpr$ExprBuffer.genApplyMethod(NirGenExpr.scala:1556)
...
```
I suppose it is related to this issue scala-native/scala-native#202

So I'll change it to `val`

Reply:
Interesting. /me flushing down `getopt.h` as candidate for #59. 😭 

I took a quick look at https://github.com/sjrd/scala-js-ts-importer/blob/master/samples/modifiers.d.ts.scala as inspiration on how to translate variables and it uses `val` for const and `def` for readonly members. @sjrd if you have time maybe you can enlighten us on the choices made for the Typescript importer.

In the standard Scala Native bindings, I see [`def stdin`](https://github.com/scala-native/scala-native/blob/master/nativelib/src/main/scala/scala/scalanative/native/stdio.scala#L92), not sure if this is because it calls a method.

Reply:
The difference between `val` and `def` is that `val` is *stable*, i.e., is guaranteed to always return the same value for the same receiver. `def` can return different values as time passes. This corresponds to `const` versus `readonly`.

Reply:
Okay, then it would be better to have `val` for `const` values and `def` for all other

Reply:
I agree. Thanks for your input @sjrd !